### ISSUE TITLE
Wrong tiles after map source change

### ISSUE THREAD

**Issue by ManuelRiegler** [route-me/route-me#183]
If you change the RMMapContents' tile source at runtime it happens that tiles from the "old" tile source are displayed and it ends up in a "wonderful" map tile mix. It looks like that the problem is somewhere at the DB (sql) cache, cause tiles are stored to the wrong DB cache.

There have been already discussions, but since that the problem still exists:
https://github.com/route-me/route-me/issues/2
http://www.mailinglistarchive.com/route-me-map@googlegroups.com/msg00403.html

removeAllCachedTiles is not really an option and  [[NSURLCache sharedURLCache] removeAllCachedResponses] right before tile source change made it better, but does not fix the problem.

**Comment by vyskocil** [comment]
Yes this bug is still there, if I remember well the culprit is the iOS NSURLrequest cache not being cleared.
A workaround is to use something like that after changing the tile source : 

```
// workaround to update display
[self.tilesView moveBy:CGSizeMake(640,960)];
[self.tilesView moveBy:CGSizeMake(-640,-960)]; 
```

**Comment by ManuelRiegler** [comment]
the problem is that somehow tiles are stored into the wrong DB. once stored in the wrong DB, a view "refresh", like the above workaround does not help.

I put some log output into "-(RMTileImage *) tileImage: (RMTile) tile" method in the RMCachedTileSource class to see which tileSource is called and to which DB the tiles are stored. I made map changes like Alpstein -> OCM -> GoogleMap -> GoogleSatellite ...always correct unit GoogleSatellite!!  at GoogleSatellite the tiles were first stored correctly to its corresponding DB, but right after the tile source changed "MAGICALLY" to Alpstein and stored the tiles also in this DB!

I really dont know why!
